### PR TITLE
a bit cleaner definition of test config spaces

### PR DESCRIPTION
**Pull request by motus**

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def test_mock_optimization_loop(mock_env_no_noise: MockEnv, mock_opt: MockOptimi
         "vmSize": "Standard_B2ms",
         "idle": "halt",
         "kernel_sched_migration_cost_ns": 117026,
-        "kernel_sched_latency_ns": 149827700,
+        "kernel_sched_latency_ns": 149827706,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py:76]: These really should be getting returned on the quantization boundary, no?

File: mlos_core/mlos_core/spaces/converters/util.py [hunk 2]
```diff
@@ -58,16 +63,23 @@ def monkey_patch_hp_quantization(hp: Hyperparameter) -> None:
...
         ConfigSpace to patch.
+
+    Returns
+    -------
+    cs : ConfigurationSpace
+        Patched ConfigSpace.
     """
     for hp in cs.values():
         monkey_patch_hp_quantization(hp)
+    return cs
```
Comment (bpkroth) [inline mlos_core/mlos_core/spaces/converters/util.py:86]: Thought about doing that too.  Works for me :)

File: mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py:76]: The proper quantization boundary here should be something like `100000000`..

The problem is that neither Mock Optimizer nor FLAML respect quantization (or distributions, for that matter). MockOptimizer just samples uniformly from the Tunables' range and does not use ConfigSpace at all; For FLAML we have a very limited conversion from ConfigSpace to FLAML hyperparameters that just ignores all that info (and I don't know if FLAML has any support for quantization etc. internally)

I still have some faint hope for SMAC3, but I cannot figure out why it does not call `ConfigSpace.sample_configuration()` like I think it should - maybe it also taps directly into the hyperparameters' ranges etc. and uses ConfigSpace only to validate the proposed configuration

**Review by bpkroth — APPROVED** [review]

File: mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/optimizers/toy_optimization_loop_test.py:76]: Yeah, I see now.  Might need to implement quantization in the mlos_bench Tunables layer, but that might interact poorly with the underlying optimizer (e.g., register a quantized version of the actual values suggested).

**Pull request merged**